### ISSUE TITLE
Documentation for starting an application inside a scala console is incorrect for Play 2.4.x

### ISSUE THREAD

**Issue by tomdepplito** [playframework/playframework#4593]
The documentation (https://www.playframework.com/documentation/2.4.x/PlayConsole) still uses `new play.core.StaticApplication(new java.io.File("."))` to start an application. However, it looks like `StaticApplication` has been removed in Play 2.4. 

Is there a new preferred way to start an application or should the documentation be updated to something like `play.api.Play.start(application)` (after manually creating an instance of `Application`)?

**Comment by alternegro** [comment]
@tomdepplito how are you currently starting an application in the console?
I've tried using `val application = new GuiceApplicationBuilder().in(Environment(new java.io.File("."), this.getClass.getClassLoader, Mode.Dev)).build` to no avail.
Thanks

**Comment by jroper** [comment]
@tomdepplito https://www.playframework.com/documentation/2.4.x/ScalaEmbeddingPlay

**Comment by alternegro** [comment]
@jroper what if you need to start the application in the console so you can interact with its component?
I have the ff in my build.sb but still I still get a "No started application exception".
````
initialCommands in console += """
  import play.api._
  import play.api.inject.guice.GuiceApplicationBuilder
  val application = new GuiceApplicationBuilder().in(Environment(new java.io.File("."), this.getClass.getClassLoader, Mode.Dev)).build
  import play.api.Play.current
  import play.api.libs.json._
  import scala.concurrent._
  import scala.concurrent.duration._
  import ExecutionContext.Implicits.global
  import models._
  import services._
"""
```

**Comment by gilbertw1** [comment]
After mucking around a bit I came up with this:

```scala
import play.api._
val env = Environment(new java.io.File("."), this.getClass.getClassLoader, Mode.Dev)
val context = ApplicationLoader.createContext(env)
val loader = ApplicationLoader(context)
val app = loader.load(context)
Play.start(app)
import Play.current
```

That got everything working in the console like I would've expected from StaticApplication. Hope that helps!

**Comment by alternegro** [comment]
@gilbertw1 That did the job. Golden! Thanks

**Comment by tomdepplito** [comment]
@jroper I arrived at a similar solution to what @gilbertw1 suggested. I'm wondering if this is now the preferred way to start an application in the Scala console? If so, I would like to update the documentation to reflect this.

**Comment by jroper** [comment]
Absolutely, the docs should be updated.

**Comment by tavlima** [comment]
If I need to start a Play app inside my main method, how am I supposed to do it in 2.4.x? I used to use play.api.{ApplicationLoader, Environment and Mode} to do that, but it's not compiling anymore, in 2.4.x...

**Comment by gamache** [comment]
:+1: I too ran into this problem.

**Comment by lennartdiedrich** [comment]
@tavlima, have you found a solution? I am struggling with the same problem...

**Comment by marcospereira** [comment]
I think this issue was fixed by #4854. You guys can also check the documentation about how to embed Play inside your application:

https://www.playframework.com/documentation/2.4.x/JavaEmbeddingPlay
https://www.playframework.com/documentation/2.4.x/ScalaEmbeddingPlay

**Comment by nafg** [comment]
On Thu, Jun 4, 2015, 3:16 PM Bryan Gilbert <notifications@github.com> wrote:

After mucking around a bit I came up with this:

import play.api._ val env = Environment(new java.io.File("."),
this.getClass.getClassLoader, Mode.Dev) val context =
ApplicationLoader.createContext(env) val loader =
ApplicationLoader(context) val app = loader.load(context) Play.start(app)
import Play.

current

Hmm... ApplicationLoader gives you the context and then you have to give it
back twice?

That got everything working in the console like I would've expected from
StaticApplication. Hope that helps!

—
Reply to this email directly or view it on GitHub
<https://github.com/playframework/playframework/issues/4593#issuecomment-109016765>.<img
src="
https://ci6.googleusercontent.com/proxy/YzhWnuciAataYqakVCGDajfcv_NRhtZ1Zg1Z90x1LkS_5fI3IudijNnQS3rddzsEfpW5CnBCQUmvakvw_U_TbXNAcMUOnZ_e-XZ4sW1qGqEA2O2R0Dr9L4cA4f-pKCv7KAhADpqMMlAt4zLOWCggF21V2uuLMA=s0-d-e1-ft#https://github.com/notifications/beacon/AAGAUPbImGC0iUdT_kSjXa77Ugb-D0Wiks5oQJuDgaJpZM4Ey_3m.gif
">
